pandas/core/frame.py

```diff
@@ -2596,7 +2601,27 @@ def transpose(self, *args, **kwargs):
         dtype: object
         """
         nv.validate_transpose(args, dict())
-        return super().transpose(1, 0, **kwargs)
+        # construct the args
+
+        if self._is_homogeneous_type and is_extension_array_dtype(self.iloc[:, 0]):
+            dtype = self.dtypes.iloc[0]
+            arr_type = dtype.construct_array_type()
+            values = self.values
```

Comment:
IIRC the single-column case could be done without this casting.  think its worth special-casing?

Reply:
I don't think it can be done without casting in general. We'll still need to reshape the `(N, 1)` to `(1, N)`.

Reply:
im pretty sure i did this in the other PR, something like:

```
values = self._data.blocks[0].values
new_vals = [values[[n]] for n in range(len(values))]
```

(of course, if we had 2D EAs...)

Reply:
So, we can avoid casting to an ndarray by making ``N * P`` length-1 `__getitem__` calls, which makes `N * P` extension arrays, which are concatenated into `N` final EAs.

My prior expectation is that converting to an ndarray and doing `__getitem__` on that will be faster, and should have roughly the same amount of memory usage.

Reply:
what is `P` here?  in the end this probably isn't worth bikeshedding (except to add to the pile of "reasons why EAs should support 2D")

Reply:
Number of columns.

Reply:
right, i was specifically referring to single-column

Reply:
I'm OK with not-doing this optimization here, just want to make sure we're on the same page about what the available optimization is